internal/proxyservice/proxy_service.go

```diff
@@ -34,8 +36,51 @@ type proxyServices struct {
 	runningServices []internal.Service
 }
 
+func (s *proxyServices) validateRequestedPluginsExist() error {
+	allPlugins := []string{"ssh", "ssh-agent"}
+	for pluginName := range s.availPlugins.TCPPlugins() {
+		allPlugins = append(allPlugins, pluginName)
+	}
+	for pluginName := range s.availPlugins.HTTPPlugins() {
+		allPlugins = append(allPlugins, pluginName)
+	}
+
+	s.logger.Infof(
+		"Validating services against available plugins: %s",
+		strings.Join(allPlugins, ","),
+	)
+
+	allPluginsMap := map[string]bool{}
+	for _, p := range allPlugins  {
+		allPluginsMap[p] = true
+	}
+
+	errors := validation.Errors{}
+	for _, service := range s.config.Services {
+		connectorExists := allPluginsMap[service.Connector]
+		if !connectorExists {
+			errors[service.Name] = fmt.Errorf(
+				`missing service connector "%s"`,
+				service.Connector)
```

Comment:
nit: I noticed you've been sometimes putting closing parens on same line, and sometimes on their own line (eg, line 51 above).  I think we should be consistent about this one way or the other.  I don't care too much which way, but it seems like we've been doing own line mostly in the code base, and it seems more consistent with Go style generally (eg, closing brace on own line).  Whereas the above feels more python-y.